framework/auth/core.py

```diff
@@ -848,6 +854,19 @@ def _get_unconfirmed_email_for_token(self, token):
 
         return verification['email']
 
+    def clean_email_verifications(self, **kwargs):
+        given_token = kwargs.get('given_token')
+        email_verifications = deepcopy(self.email_verifications)
+        for token in self.email_verifications:
+            try:
+                self.get_unconfirmed_email_for_token(token)
+            except (KeyError, ExpiredTokenError):
+                email_verifications.pop(token)
+                break
```

Comment:
Why the `break`? Won't this mean that only one email can get cleaned?

Would be good to have tests for this method.